### ISSUE TITLE
Laravel 5.2 incompatibility due to symfony/event-dispatcher

### ISSUE THREAD

**Issue by knicola** [ringcentral/ringcentral-php#19]
There's an incompatibility between laravel 5.2 and ringcentral-php due to symfony/event-dispatcher ^2.7. Laravel currently uses v3.0.1

Would it be possible for ringcentral-php to pump the version up to 3.0.1 also?

**Comment by kirill-konshin** [comment]
Yes, I will look into it.

**Comment by kirill-konshin** [comment]
I have pushed updated `composer.json` to develop branch.

**Comment by kirill-konshin** [comment]
3.* requires PHP 5.5.9 and currently RC PHP SDK supports PHP 5.3.

**Comment by knicola** [comment]
I'm currently running 5.5.9 and I haven't noticed any issues with the SDK.

**Comment by kirill-konshin** [comment]
RC SDK is fully compatible with 5.5.9.

The issue is that some of our clients still use PHP 5.3 so we can't upgrade our dependencies right away. We have plans of discarding support of 5.3 and 5.4 in nearest future, within this process I will also upgrade all the dependencies, stay tuned.

**Comment by kirill-konshin** [comment]
I have pushed the 2.0.0 candidate to develop branch. You can start using it by having `dev-develop` as SDK version in your `composer.json`. Contact me if you'll have any issues.

**Comment by bdeanindy** [comment]
I received the following output while installing the SDK via composer in PHP 5.6:

```
benjamin.dean@LMRC3973:~/ApacheSites/ringcentralPHP56 (HEAD: ref: refs/heads/master) $ php composer.phar require ringcentral/ringcentral-php
Using version ^1.1 for ringcentral/ringcentral-php
./composer.json has been created
Loading composer repositories with package information
Updating dependencies (including require-dev)
  - Installing symfony/event-dispatcher (v2.8.4)
    Downloading: 100%         

  - Installing pubnub/pubnub (3.8.1)
    Downloading: 100%         

  - Installing psr/http-message (1.0)
    Downloading: 100%         

  - Installing ringcentral/psr7 (1.2.1)
    Downloading: 100%         

  - Installing ringcentral/ringcentral-php (1.1.1)
    Downloading: 100%         

symfony/event-dispatcher suggests installing symfony/dependency-injection ()
symfony/event-dispatcher suggests installing symfony/http-kernel ()
Writing lock file
Generating autoload files
```

Once I switched the version in `composer.json` to be 'dev-develop' then I was able to install via composer without any notifications of the symfony/event-dispatcher suggestions.

**Comment by kirill-konshin** [comment]
Is the issue still reproduced on version 2.0.0?

**Comment by izzyp** [comment]
@kirill-konshin 

Is RC v2 only compatible with version symfony/event-dispatcher v3.0?

We're using Laravel 5.1 on a project which requires symfony/event-dispatcher ^2.7 making RC v2 incompatible.

Would RC v2 potentially work with either "^2.7|^3.3" ?

If so, can you update the dependancy in composer to reflect compatibility with either version.
"symfony/event-dispatcher": "^2.7|^3.3"

**Comment by anilkumarbp** [comment]
@izzyp 

I have created a branch named `laravel-5.1` on my [Fork](https://github.com/anilkumarbp/ringcentral-php) and have updated the dependency in the composer for the package : 

"symfony/event-dispatcher": "^2.7|^3.3"

Can you validate, please? Just copy this in your composer.json 

```
    "repositories": [
        {
            "type": "git",
            "url": "https://github.com/anilkumarbp/ringcentral-php"
        }
    ],
    "require": {
        "php": ">=5.5.9",
        "laravel/framework": "5.1.*",
        "ringcentral/ringcentral-php": "dev-laravel-5.1"
    }
```
I would need to vaildate if the same works on 5.2 and does not break the backward compatibility.

**Comment by izzyp** [comment]
@anilkumarbp 

Yup - working fine.

Thanks!

**Comment by kirill-konshin** [comment]
Released 2.1.1
